owncloudComLibrary/src/main/java/com/owncloud/android/lib/resources/appregistry/CreateRemoteFileWithAppProviderOperation.kt

```diff
@@ -0,0 +1,108 @@
+import com.owncloud.android.lib.common.operations.RemoteOperationResult.ResultCode
+import com.squareup.moshi.Json
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.JsonClass
+import com.squareup.moshi.Moshi
+import okhttp3.FormBody
+import okhttp3.RequestBody
+import timber.log.Timber
+import java.net.URL
+import java.util.concurrent.TimeUnit
+
+class CreateRemoteFileWithAppProviderOperation(
+    private val createFileWithAppProviderEndpoint: String,
+    private val parentContainerId: String,
+    private val filename: String,
+) : RemoteOperation<String>() {
+
+    override fun run(client: OwnCloudClient): RemoteOperationResult<String> {
+        return try {
+
+            val createFileWithAppProviderRequestBody = CreateFileWithAppProviderParams(parentContainerId, filename)
+                .toRequestBody()
+
+            val stringUrl = client.baseUri.toString() + WebdavUtils.encodePath(createFileWithAppProviderEndpoint)
+
+            val postMethod = PostMethod(URL(stringUrl), createFileWithAppProviderRequestBody).apply {
+                setReadTimeout(TIMEOUT, TimeUnit.MILLISECONDS)
+                setConnectionTimeout(TIMEOUT, TimeUnit.MILLISECONDS)
+            }
+
+            val status = client.executeHttpMethod(postMethod)
+            Timber.d("Create file $filename with app provider in folder with ID $parentContainerId - $status${if (!isSuccess(status)) "(FAIL)" else ""}")
+
+            if (isSuccess(status)) RemoteOperationResult<String>(ResultCode.OK).apply {
+                val moshi = Moshi.Builder().build()
+                val adapter: JsonAdapter<CreateFileWithAppProviderResponse> = moshi.adapter(CreateFileWithAppProviderResponse::class.java)
+
+                data = postMethod.getResponseBodyAsString()?.let { adapter.fromJson(it)!!.fileId }
+            }
+            else RemoteOperationResult<String>(postMethod).apply { data = "" }
+
+        } catch (e: Exception) {
+            val result = RemoteOperationResult<String>(e)
+            Timber.e(e, "Create file $filename with app provider in folder with ID $parentContainerId failed")
+            result
+        }
+    }
+
+    private fun isSuccess(status: Int) = status == HttpConstants.HTTP_OK
+
+    data class CreateFileWithAppProviderParams(
+        val parentContainerId: String,
+        val filename: String,
+    ) {
+        fun toRequestBody(): RequestBody =
+            FormBody.Builder()
+                .add(PARAM_PARENT_CONTAINER_ID, parentContainerId)
+                .add(PARAM_FILENAME, filename)
+                .build()
+
+        companion object {
+            const val PARAM_PARENT_CONTAINER_ID = "parent_container_id"
+            const val PARAM_FILENAME = "filename"
+        }
+    }
+
+    @JsonClass(generateAdapter = true)
+    data class CreateFileWithAppProviderResponse(
+        @Json(name = "file_id")
+        val fileId: String,
+    )
+
+    companion object {
+        private const val TIMEOUT: Long = 5_000
```

Comment:
Would be great to unify timeouts across similar operations.

What do you think about that? Do you think it is feasible? 🤔

Reply:
Yes, I think that's a good idea, but I see there are lots of timeout variables with different values, and some of them are defined as seconds and others as milliseconds (even sometimes I see 10_000 seconds, I think that's not correct). We would need to review this but I think it's out of the scope of this PR. Feel free to open a new issue with this if you want 🍻